packages/gatsby/src/utils/api-node-helpers-docs.js

```diff
@@ -0,0 +1,293 @@
+};
+
+/** */
+const GatsbyCache = {
+  /**
+   * @param {string} key
+   * @returns {Promise<any>}
+   * @example
+   * const value = await cache.get(`unique-key`)
+   */
+  get: true,
+
+  /**
+   * @param {string} key
+   * @param {any} value
+   * @returns {Promise<any>}
+   * @example
+   * await cache.set(`unique-key`, value)
+   */
+  set: true,
+};
+
+/** */
+const GatsbyTracing = {
+  /**
+   * Global tracer instance. Check
+   * [opentracing Tracer documentation](https://opentracing-javascript.surge.sh/classes/tracer.html)
+   * for more details.
+   * @type {Opentracing.Tracer}
+   */
+  tracer: true,
+
+  /**
+   * Tracer span representing API run. Check
+   * [opentracing Span documentation](https://opentracing-javascript.surge.sh/classes/span.html)
+   * for more details.
+   * @type {Opentracing.Span}
+   */
+  parentSpan: true,
+
+  /**
+   * @callback GatsbyTracing.StartSpan
+   * @param {string} spanName name of the span
+   * @returns {Opentracing.Span}
+   */
+
+  /**
+   * Convience function to start span using API run span as parent.
+   * @type {GatsbyTracing.StartSpan}
+   * @example
+   * exports.sourceNodes = async ({ actions, tracing }) => {
+   *   const span = tracing.startSpan(`foo`)
+   *
+   *   // Perform any span operations. E.g add a tag to your span
+   *   span.setTag(`bar`, `baz`)
+   *
+   *   // Rest of your plugin code
+   *
+   *   span.finish()
+   * }
+   */
+  startSpan: true,
+};
+
+/** */
+const GatsbyNodeHelpers = {
+  /**
+   * Key-value store used to persist results of time/memory/cpu instensive
+   * tasks. All functions are async and return promises.
+   * @type {GatsbyCache}
+   */
+  cache: true,
+
+  /**
+   * Get cache instance by name - this should only be used by plugins that
+   * accept subplugins.
+   * @param {string} id Test
+   * @returns {GatsbyCache} See [`cache`](#cache) section for reference.
+   */
+  getCache: true,
+
+  /**
+   * Create stable content digest from string or object. Useful to set
+   * `internal.contentDigest` field on nodes to make sure that Gatsby
+   * either invalidates cached node and GraphQL query results depending
+   * on that if data changed or reuse cached query results if data didn't
+   * change.
+   * @param {(string|object)} input
+   * @returns {string} Hash string
+   * @example
+   * const node = {
+   *   ...nodeData,
+   *   internal: {
+   *     type: `TypeOfNode`,
+   *     contentDigest: createContentDigest(nodeData)
+   *   }
+   * }
+   */
+  createContentDigest: true,
+
+  /**
+   * All action creators wrapped with a dispatch.
+   *
+   * See [`actions`](/docs/actions/) reference.
+   * @type {Actions}
+   * @deprecated Will be removed in gatsby 3.0. Use [actions](#actions)
+   * instead.
+   */
+  boundActionCreators: true,
+
+  /**
+   * All action creators wrapped with a dispatch.
+   *
+   * See [`actions`](/docs/actions/) reference.
+   * @type {Actions}
+   */
+  actions: true,
+
+  /**
+   * @param {Node} node
```

Comment:
Missing description? We can add this in a follow up PR.

Reply:
I remember now why I skipped this - we have this logic:
https://github.com/gatsbyjs/gatsby/blob/f7a6bb5d9615e3952827a8736039eba9aa372e28/packages/gatsby/src/db/nodes.js#L22-L50

But I didn't find documentation for implementing custom `loadNodeContent` function in plugins (not using it, but providing it in case `internal.content` isn't set). Example of this is in `gatsby-source-filesystem`
https://github.com/gatsbyjs/gatsby/blob/f7a6bb5d9615e3952827a8736039eba9aa372e28/packages/gatsby-source-filesystem/src/index.js#L3-L10

This allows to lazy load content (so we don't read file content from f.e. images, because no transformer actually need that as a string)

I will just use `Get content for a node from the plugin that created it.` so it at least have some description, but this needs to be documented further (probably separate PR)

Reply:
Added very basic description in https://github.com/gatsbyjs/gatsby/pull/12087/commits/5d351ea79d9011339689427c6d3f2293d109f02f - need more work, but don't want to block this PR